functorch/test/test_ops.py

```diff
@@ -1221,6 +1221,166 @@ def reference(primals, cotangents, primals_tangents, cotangents_tangents):
+        xfail('sparse.sampled_addmm'),  # RuntimeError: Sparse CSR tensors do not have strides
+        xfail('svd_lowrank'),  # calls random op
+        xfail('symeig'),  # Forward AD not implemented and no decomposition
+        xfail('take'),  # vmap: inplace into regular tensor
+        xfail('to'),  # RuntimeError: required rank 4 tensor to use channels_last format
+        xfail('to_sparse'),  # Forward AD not implemented and no decomposition
+        xfail('view_as_complex'),  # RuntimeError: Tensor must have a last dimension with stride 1
+    }))
+    @ops(op_db + additional_op_db, allowed_dtypes=(torch.float,))
+    @toleranceOverride({torch.float32: tol(atol=1e-04, rtol=1e-04)})
+    @opsToleranceOverride('TestOperators', 'test_vmapjvpvjp', (
+        tol1('linalg.svd',
+             {torch.float32: tol(atol=5e-04, rtol=5e-04)}),
+        tol1('linalg.householder_product',
+             {torch.float32: tol(atol=5e-04, rtol=5e-04)}),
+        tol1('linalg.multi_dot',
+             {torch.float32: tol(atol=5e-04, rtol=5e-04)}),
+        tol1('svd',
+             {torch.float32: tol(atol=5e-04, rtol=5e-04)}),
+    ))
+    # linalg.svd - manual tolerance
+    # svd
+    def test_vmapjvpvjp(self, device, dtype, op):
+        # Since we test `jvpvjp` seperately,
+        # in this we just check that vmap of `jvpvjp`
+        # is correct.
+        if not op.supports_autograd:
+            self.skipTest("Skipped! Autograd not supported.")
+            return
+
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+
+        # TODO: test in-place
+        if is_inplace(op, op.get_op()):
+            self.skipTest("Skipped! NYI: inplace-testing not supported.")
+            return
+
+        for sample in samples:
+            fn, primals = normalize_op_input_output(op, sample)
+            result = fn(*primals)
+            cotangents = tree_map(lambda x: torch.randn_like(x), result)
+
+            primals_tangents = tree_map(lambda x: torch.randn_like(x), primals)
+            cotangents_tangents = tree_map(lambda x: torch.randn_like(x), cotangents)
+
+            if isinstance(primals[0], torch.Tensor) and primals[0].numel() == 0:
+                # typically the first primal arg is the input. If the input has no elements, we will typically run
+                # into an issue of "Expected Tensor but got None"
+                continue
+
+            def push_vjp(primals, cotangents):
+                _, vjp_fn = vjp(fn, *primals)
+                return vjp_fn(cotangents)
+
+            args, spec = tree_flatten(((primals, cotangents), (primals_tangents, cotangents_tangents)))
+
+            def jvp_of_vjp(*args):
+                (primals, tangents) = tree_unflatten(args, spec)
+                primals_out, tangents_out = jvp(push_vjp, primals, tangents)
+
+                if isinstance(primals_out, torch.Tensor):
+                    return (primals_out, tangents_out)
```

Comment:
This case shouldn't be necessary -- tree_flatten(Tensor) returns a list containing the single tensor